### ISSUE TITLE
literal routes problem

### ISSUE THREAD

**Issue by boedociclon** [coolcsn/CsnAuthorization#15]
Hi and thank you for these great modules!
I have installed skeleton application and all csn modules and everything works fine.
But there is a little issue:

I created customAction() inside Application\Controller\IndexController;
then I added a record in privilege table for resource Application\Controller\Index and name custom;
If I request route /application/index/custom everything works fine

But I need literal route for this action so I created it: /mycustomroute pointing to same action
Now if I request route /mycustomroute authorization fails, while segment route still works fine.

Can you please help me?
Thank you very much.
Regards.

**Comment by wingman007** [comment]
Hello,

Authorization is a tricky one! :)
We have not tested very well CsnAuthorization when it works with database. I am using a file acl.global.php instead.
You can turn CsnAuthorization off temporarily just to make sure your routes are correct (comment the module CsnAuthorization in application.config.php).
1. Make sure the route leads exactly to the expected module controller and action.
2. In config/autoload/acl.global.php make sure you have given the right privileges. First try to allow the action for everyone. Make sure you have a record in acl.global.php such as: 
...
'Application\Controller\Index' => array(
                    'custom'   => 'guest',
                                         ..
                ),
...

I hope this helps!
Stoyan

**Comment by boedociclon** [comment]
Hi and thanks for your answer.
I finally figure out the solution for this issue.
Fisrt of all: database configuration of CsnAuthorization Module works as well as file one!
The problem was about definition of custom literal routes in Application module configuration file:
actually I had method myCustomAction in Application\Controller\Index and I defined literal route /custom.
Issue is about definition of the action of the route: 
both 
'action' => 'myCustom'
and 
'action' => 'my-custom'
works fine without CsnAuthorization Module, but only the second one works with CsnAuthorization Module.

In conclusion this is correct way to use custom literal routes with CsnAuthorization Module:
never use camelCase notation for action definitions!

I hope this may help someone with same problem.

Thanks again for your great job with these useful modules.
Regards.
Davide.

**Comment by wingman007** [comment]
Thank you very much for sharing your solution with us!
